src/spglib.h

```diff
@@ -341,35 +358,34 @@ SpglibMagneticSpacegroupType spg_get_magnetic_spacegroup_type_from_symmetry(
     const int *time_reversals, const int num_operations,
     const double lattice[3][3], const double symprec);
 
-/* Return exact number of symmetry operations. This function may */
-/* be used in advance to allocate memory space for symmetry */
-/* operations. */
+SPG_DEPRECATED("FIXME: What to replace with?")
```

Comment:
The document says "This function may be used in advance to allocate memory space for symmetry operations".
@atztogo does this function need to be deprecated?

Reply:
No. This is somehow necessary to use with `spg_get_symmetry`. We don't know how many operations are returned. This function exists for this purpose. `spg_get_dataset` is unnecessary to know the number of operations returned in advance, but this requires `spg_free_dataset` to avoid memory leak.

By the way, do we have any solution about this in modern C?

Reply:
Walk me through this one, where does the user this afterwards?

My reading here is that the user should be in charge of calling get_dataset, get everything they need, and call free_dataset, handling all of the memory leak issues themselves.

> By the way, do we have any solution about this in modern C?

Modern C, no. The only modern approach would be a C++ interface where all of these issues are mute.

Note, even if such functions are not planned for removal, I would still suggest to mark them as deprecated, in order to indicate to the user, and especially those writing binding APIs, that this should not be used or exposed, instead, they should take ownership of the dataset object and deconstruct when needed.

Reply:
I agree your opinion, but users may have different opinions. However number of users who use C-API directly may be few, so we may consider to make it deprecated. How do you think @lan496?

Reply:
I also think there will be few users using this function. Then, the deprecation would be reasonable.